packages/mobile/src/tokens/saga.ts

```diff
@@ -313,6 +313,22 @@ export async function fetchTokenBalancesForAddress(
   return response.data.userBalances.balances
 }
 
+export function* updateTokenBalances(tokens: StoredTokenBalances) {
+  const tokenBalanceBefore = yield select(totalTokenBalanceSelector)
+  const isAccountFundedBefore = tokenBalanceBefore.gt(DOLLAR_MIN_AMOUNT_ACCOUNT_FUNDED)
+
+  yield put(setTokenBalances(tokens))
```

Comment:
i *think* that the `yield put` will wait for the redux state to change before moving to the next part.

there's some pretty complex stuff going on in the `totalTokenBalanceSelector` which is nicely memoised right now so this was the most straightforward way i could think of to reuse that logic.